### DISCUSSION TITLE
Pythonnet 3.0 release date?

### DISCUSSION THREAD

**Discussion by saroad2** [pythonnet/pythonnet#1694] [General]
Hey *Pythonnet* team.

I use the *Pyhonnet* library in multiple projects such as [Toga](https://github.com/beeware/toga) and [Eddington-GUI](https://github.com/EddLabs/eddington-gui). I'm in a situation that In order to develop new features for my projects I need to use the latest *Pythonnet* version, which is an alpha version of *Pythonnet* 3.0.

Even though this alpha version is helpful, I don't want to rely on that version since alpha versions are not stable API-vise by definition.

Could you give an estimation when the official version will be released? it seems like there are always 3-5 issues preventing from releasing the actual version.

Thanks.

**Comment by filmor** [comment]
I don't want to release while we have known crashes. We currently have those in https://github.com/pythonnet/pythonnet/issues/1642 and https://github.com/pythonnet/pythonnet/issues/1683. Once these are sorted out, we'll create a release candidate. We currently don't have any plans to adjust the APIs further (both in Python and C#) before the release, and if we do, we'll call it out on the respective changelog.

**Reply by saroad2** [reply]
It seems that the issues stated above have been solved.
Anything else preventing from releasing *Pythonnet 3.0*?

**Reply by surdouski** [reply]
I would also like to know the answer to this. The codebase appears to be rapidly changing, making it difficult for me to work on my current project -- if there was a stable release I'd feel more comfortable than working off a master branch install, as I want to be confident future releases don't undergo another code reformatting and restructuring.

**Reply by lostmsu** [reply]
@surdouski while the code is moving around the last breaking change was on March 3.
